lib/iris/fileformats/netcdf/saver.py

```diff
@@ -2387,7 +2275,7 @@ def _increment_name(self, varname):
 
         return "{}_{}".format(varname, num)
 
-    def _lazy_stream_data(self, data, fill_value, fill_warn, cf_var):
+    def _lazy_stream_data(self, data, cf_var):
```

Comment:
c85c7bcf3b6ac4b91be04b5af81f8dcc46823c1c

Reply:
@trexfeathers How's about adopting the following pattern ...

```python
from __future__ import annotations

...

if typing.TYPE_CHECKING:
    from numpy.typing import ArrayLike

    class VariableEmulator(typing.Protocol):
        """Duck-type-hinting for a ncdata object.

        https://github.com/pp-mo/ncdata
        """
        _data_array: ArrayLike

    CFVariable: typing.TypeAlias = typing.Union[_thread_safe_nc.VariableWrapper, VariableEmulator]
```

Then using the ```CFVariable``` type-alias (or however you want to call it) later as:

```python
def _lazy_stream_data(
        self,
        data: ArrayLike,
        cf_var: CFVariable,
    ) -> None:
```

which make the code more readable.

Reply:
9a186b8c930c3a6eb7f3fcf91106eb9828ffabdc

Since I wasn't importing anything specifically for this, I decided I didn't need the `if typing.TYPE_CHECKING:` pattern.